contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3012,18 +3012,29 @@ static void bbf_ProcessUtility(PlannedStmt *pstmt,
 			}
 			break;
 		case T_RenameStmt:
+		{
+			if (sql_dialect == SQL_DIALECT_TSQL)
 			{
-				if (prev_ProcessUtility)
-					prev_ProcessUtility(pstmt, queryString, readOnlyTree, context, params,
-							queryEnv, dest, qc);
-				else
-					standard_ProcessUtility(pstmt, queryString, readOnlyTree, context, params,
-							queryEnv, dest, qc);
-
-				check_extra_schema_restrictions(parsetree);
+				RenameStmt *stmt = (RenameStmt *) parsetree;
 
+				PG_TRY();
+				{
+					if (prev_ProcessUtility)
+						prev_ProcessUtility(pstmt, queryString, readOnlyTree, context,
+											params, queryEnv, dest, qc);
+					else
```

Comment:
at least this function is wrong from my perspective. 

it should be 
				if (prev_ProcessUtility)
					prev_ProcessUtility(pstmt, queryString, readOnlyTree, context, params,
							queryEnv, dest, qc);
				else
					standard_ProcessUtility(pstmt, queryString, readOnlyTree, context, params,
							queryEnv, dest, qc);
                check_extra_schema_restrictions			
                 if (dialect = tsql) 
			      rename_update_bbf_catalog
                 return;